src/Tribe/iCal.php

```diff
@@ -183,6 +204,69 @@ public function do_ical_template() {
 		}
 	}
 
+	/**
+	 * Generates the iCal file
+	 *
+	 * @param int|null $post If you want the ical file for a single event
+	 * @param boolean  $echo Whether the content should be echoed or returned
+	 *
+	 * @return string
+	 */
+	public function generate_ical_feed( $post = null, $echo = true ) {
+		$this->post = $post;
+		$this->events = $this->get_event_posts();
+		$content = $this->get_content();
+
+		if ( $echo ) {
+			$this->set_headers();
+			tribe_exit( $content );
+		}
+
+		return $content;
+	}
+
+	/**
+	 * Get an array with all the Events to be used to process the *.ics file
+	 *
+	 * @since TBD
+	 *
+	 * @return array|null
+	 */
+	protected function get_event_posts() {
+		if ( $this->post ) {
```

Comment:
I'm curious why we are using the  elseif instead of just using the ifs, as each one has a return.

Reply:
No real reason, but primarily as this one was the original code this change just moved this piece into a function. But I think is a great point which will make this function so much easier to understand.